internal/verifier/verifier.go

```diff
@@ -124,36 +124,26 @@ func chdir(dir string) {
 	}
 }
 
-// AmberProvenanceMetadataVerifier verifies Amber provenances by comparing the
+// ProvenanceMetadataVerifier verifies provenances by comparing the
 // content of the provenance predicate against a given set of expected values.
-type AmberProvenanceMetadataVerifier struct {
-	provenanceFilePath string
+type ProvenanceMetadataVerifier struct {
+	Got  *slsa.ValidatedProvenance
+	Want ProvenanceIR
 	// TODO(#69): Add metadata fields.
 }
 
-// Verify verifies a given Amber provenance file by checking its content
-// against the expected values specified in this
-// AmberProvenanceMetadataVerifier instance. Returns an error if any of the
-// values is not as expected. Otherwise returns nil, indicating success.
+// Verify verifies a given provenance file by checking its content against the expected values
+// ProvenanceMetadataVerifier instance.
 // TODO(#69): Check metadata against the expected values.
-// TODO(#126): Refactor and separate verification logic from the logic for reading the file.
```

Comment:
Can #126 be closed now? Or are there more references to it?

Reply:
Yes, there are: The `ReproducibleProvenanceVerifier`.